server/src/main/java/org/elasticsearch/action/search/SearchQueryThenFetchAsyncAction.java

```diff
@@ -74,13 +78,36 @@
         boolean hasFetchPhase = request.source() == null ? true : request.source().size() > 0;
         progressListener.notifyListShards(SearchProgressListener.buildSearchShards(this.shardsIts),
             SearchProgressListener.buildSearchShards(toSkipShardsIts), clusters, hasFetchPhase);
+        if (checkMinimumVersion(shardsIts) == false) {
```

Comment:
I don't think it's valid to just throw an exception here. This logic should be moved to `AbstractSearchAsyncAction#run` that is executed by `#executePhase` with the correct try/catch (raising a phase failure in case of errors) ?